### PR TITLE
image+LibGfx: Make it possible to do actual color profile conversions from CMYK to RGB

### PR DESCRIPTION
**Pull request by nico**
The first couple commits reorganize `image` a bit. The latter commits make it read CMYK data from CMYK frames. The only thing we can currently do with CMYK data is to convert it to RGB. This was possible in PDFs for a while, but now `image` can do it to.

### CODE REVIEW

File: Userland/Utilities/image.cpp [hunk 1]
```diff
@@ -100,38 +107,36 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     if (!decoder)
         return Error::from_string_view("Failed to decode input file"sv);
 
-    auto frame = TRY(decoder->frame(frame_index)).image;
+    LoadedImage image = { TRY(decoder->frame(frame_index)).image, TRY(decoder->icc_data()) };
 
     if (move_alpha_to_rgb)
-        TRY(do_move_alpha_to_rgb(frame));
+        TRY(do_move_alpha_to_rgb(image));
```
Comment (LucasChollet) [inline Userland/Utilities/image.cpp:209]: FYI: Passing `image.bitmap` also works here

File: Userland/Utilities/image.cpp [hunk 2]
```diff
@@ -16,18 +16,37 @@
 #include <LibGfx/ImageFormats/QOIWriter.h>
 
 struct LoadedImage {
-    RefPtr<Gfx::Bitmap> bitmap;
+    Gfx::NaturalFrameFormat internal_format;
+    AK::Variant<RefPtr<Gfx::Bitmap>, RefPtr<Gfx::CMYKBitmap>> bitmap;
```
Comment (LucasChollet) [inline Userland/Utilities/image.cpp:20]: Note:
On a short term, we should name that `Variant`.
For the long term, I'm also thinking more and more about a class hierarchy for `Bitmaps` and that will be a `RefPtr<BitmapBase>`

File: Userland/Utilities/image.cpp [hunk 3]
```diff
@@ -16,18 +16,37 @@
...
+
     Optional<ReadonlyBytes> icc_data;
 };
 
 static ErrorOr<LoadedImage> load_image(RefPtr<Gfx::ImageDecoder> const& decoder, int frame_index)
 {
-    return LoadedImage { TRY(decoder->frame(frame_index)).image, TRY(decoder->icc_data()) };
+    auto internal_format = decoder->natural_frame_format();
+
+    AK::Variant<RefPtr<Gfx::Bitmap>, RefPtr<Gfx::CMYKBitmap>> bitmap = TRY([&]() -> ErrorOr<decltype(bitmap)> {
```
Comment (LucasChollet) [inline Userland/Utilities/image.cpp:29]: I kinda like this `decltype`, but at the same time I think that this is better (feel free to disagree):
`auto bitmap = []() -> ErrorOr<Variant<RefPtr<Gfx::Bitmap>, RefPtr<Gfx::CMYKBitmap>>> ...`

And again, no need for `AK`

File: Userland/Libraries/LibGfx/ICC/Profile.cpp [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
 #include <AK/Endian.h>
 #include <LibGfx/CIELAB.h>
+#include <LibGfx/CMYKBitmap.h>
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/ICC/Profile.cpp:9]: `s/convers/converts/` in the commit message

File: Userland/Utilities/image.cpp (hunk 2 above)
Comment (LucasChollet) [inline Userland/Utilities/image.cpp:20]: Also, you shouldn't need to use `AK` here.

**Review by LucasChollet — COMMENTED** [review]
What happened to you? A 14-commits PR, really?

File: Userland/Utilities/image.cpp (hunk 1 above)
Comment (nico) [reply Userland/Utilities/image.cpp:209]: In this commit, yes. But a later commit makes the function error out for cmyk images.

File: Userland/Utilities/image.cpp (hunk 2 above)
Comment (nico) [reply Userland/Utilities/image.cpp]: VSCode's autocomplete adds it :( I thought I had removed them all, thanks for catching this.

**Comment by nico** [comment]
Red CI is unrelated.

**Comment by nico** [comment]
Here's a before/after with a local .pam reader. Before:

out-before

After (`Build/lagom/bin/image -o out.png --convert-to-color-profile serenity-sRGB.icc --assign-color-profile ~/Downloads/Adobe\ ICC\ Profiles\ \(end-user\)/CMYK/USWebCoatedSWOP.icc  image-6321.pam`):

out-after

**Review by LucasChollet — APPROVED** [review]

**Pull request merged**